### PR TITLE
refactor: improve poetry packaging

### PR DESCRIPTION
**Pull request by Scrumplex**
This allows poetry-only builds (i.e. `poetry install`) and better packaging for distributions.

Additionally this PR adds Nix packaging to allow users of Nix to build and run the app locally.

### CODE REVIEW

**Comment by ShyAssassin** [comment]
this looks good to me!
could you please rebase this PR to the [features/algorithms](https://github.com/EyeTrackVR/ETVR-Backend/tree/feature/algorithms) branch as this is currently the active dev branch, i will do a full code review + testing later today!

**Comment by Scrumplex** [comment]
I will be able to reduce the flake.nix once https://github.com/nix-community/poetry2nix/pull/1803 is merged

**Review by ShyAssassin — CHANGES_REQUESTED** [review]
Everything looks good, the only thing i can comment on is that the `makefile` and `build.py` script have not been updated to reflect the backends new directory.

File: eyetrackvr_backend/main.py [hunk 1]
```diff
@@ -4,9 +4,9 @@
     os.chdir(os.path.dirname(__file__))
 
 from fastapi.staticfiles import StaticFiles
-from app.logger import setup_logger
+from .app.logger import setup_logger
 from fastapi import FastAPI
-from app.etvr import ETVR
+from .app.etvr import ETVR
```
Comment (ShyAssassin) [inline eyetrackvr_backend/main.py]: making this import relative seems to break things with import errors and prevent startup when running from main directly

Comment (Scrumplex) [reply eyetrackvr_backend/main.py]: Hmm, how are you starting the application? I am running `python -m eyetrackvr_backend.main` which will treat the folder main resides in as a package

Comment (ShyAssassin) [reply eyetrackvr_backend/main.py]: i am running it directly with `poetry run python eyetrackvr_backend/main.py` but this relative package also cannot be found after being packaged with pyinstaller

Comment (Scrumplex) [reply eyetrackvr_backend/main.py]: I see. Let me see how we can bend pyinstaller to run it as a module instead (i.e. equivalent to `poetry run python -m eyetrackvr_backend.main`)

Comment (Scrumplex) [reply eyetrackvr_backend/main.py]: Seems like pyinstaller does not support modules properly? A workaround is to have a script outside the module directory that calls main.

**Comment by Scrumplex** [comment]
Okay I changed the structure again. In my local testing, the pyinstaller binary works as well as running it manually (i.e. `poetry run eyetrackvr-backend`)

**Review by ShyAssassin — APPROVED** [review]

**Comment by ShyAssassin** [comment]
Juts did a full test with my ETVR hardware locally and everything looks good!
there seems to be a bug related to the camera process not detecting a new source address when config is changed but i do not believe this PR introduced that bug.

**Pull request merged**